src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs

```diff
@@ -70,4 +70,14 @@ public interface IFileStore
         VersionedInstanceIdentifier versionedInstanceIdentifier,
         FrameRange range,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously deletes a old DICOM file from the file store if the file exists.
+    /// </summary>
+    /// <param name="versionedInstanceIdentifier">The DICOM identifier.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A task that represents the asynchronous delete operation.</returns>
+    Task DeleteOldFileIfExistsAsync(
```

Comment:
Should this method return a flag indicate whether it was deleted?

Reply:
Successful running of this task means it is deleted. Why do you think we need a flag?

Reply:
Typically "if exists" methods return some indication of what actually happened. Did this thing actually exist and we deleted it, or was it skipped? For example, the Azure Storage SDKs include methods like this that have different results depending on whether the element was present. I think their pattern is typically like `Task<Response> DeleteIfExists(CancellationToken)` where the response object is `null` if the value didn't exist.

Reply:
We have a method `DeleteFileIfExistsAsync` which doesn't return any flag. Do you want two different implementations in the same file?

Reply:
Ah that's unfortunate. I suppose for symmetry's sake we should keep as-is, but we should be more intentional going forward